MovieSharing/APIManager.swift

```diff
@@ -21,39 +21,41 @@ class APIManager {
         case emptyData
     }
     
+    // MARK: - Public API
+    
+    func fetchVideos(url: String, completion: @escaping (Result<VideoContainer,VideoError>) -> Void){
+        
+        fetchData(url: url){ result in
+            
+            do {
+                let videoData = try result.get()
+                let videosNotNil = try JSONDecoder().decode(VideoContainer.self, from: videoData)
+                completion(.success(videosNotNil))
+                
+            } catch {
+                completion(.failure(.noDataAvailable))
+            }
+        }
+    }
+    
     // MARK: - Private API
     
     private init() {}
-
-    // MARK: - Public API
     
-    func fetchVideos(url: String, completion: @escaping (Result<[Video],VideoError>) -> Void) {
-        
+    private func fetchData(url: String, completion: @escaping (Result<Data, VideoError>) -> Void) {
         
         guard let resourceURL = URL(string: url) else {
             return
         }
         
-        let dataTask = URLSession.shared.dataTask(with: resourceURL) {
-            data, _, _ in
+        let dataTask = URLSession.shared.dataTask(with: resourceURL){
+            data, response, error in
             
             guard let jsonData = data else {
```

Comment:
- [ ] the data in this generalized method won't necessarily be in json format so consider renaming to `dataNotNil`